packages/components/src/internal/components/domainproperties/samples/SampleTypeDesigner.tsx

```diff
@@ -319,9 +324,12 @@ class SampleTypeDesignerImpl extends React.PureComponent<Props & InjectedBaseDom
     };
 
     onFinish = (): void => {
-        const { defaultSampleFieldConfig, setSubmitting } = this.props;
+        const { defaultSampleFieldConfig, setSubmitting, metricUnitProps } = this.props;
         const { model } = this.state;
-        const isValid = model.isValid(defaultSampleFieldConfig);
+
+        let metricUnitLabel = metricUnitProps ? metricUnitProps.metricUnitLabel : undefined;
+        let metricUnitRequired = metricUnitProps ? metricUnitProps.metricUnitRequired : undefined;
```

Comment:
These could be `metricUnitProps?.metricUnitLabel` and `metricUnitProps?.metricUnitRequired` instead